src/Periphery/LiFuelFeeCollector.sol

```diff
@@ -53,15 +53,17 @@ contract LiFuelFeeCollector is TransferrableOwnership {
     /// @param chainId The chain id of the destination chain
     /// @param receiver The address to send gas to on destination chain
     function collectNativeGasFees(
+        uint256 feeAmount,
         uint256 chainId,
         address receiver
     ) external payable {
         emit GasFeesCollected(
             LibAsset.NULL_ADDRESS,
             chainId,
             receiver,
-            msg.value
+            feeAmount
         );
+        msg.sender.call{ value: msg.value - feeAmount }("");
```

Comment:
Add check to validate if there is a amount to send back. e.g. the lifuel bridge sends the exact amount needed.